rise-storage.html

```diff
@@ -473,35 +486,52 @@
     },
 
     /**
-     * Fires when an error is received from the Storage request.
+     * Retry the storage request one more time if it fails and attempt to load file from cache
      */
-    _handleStorageError: function(e, resp) {
-      var fireError = true;
+    _handleStorageErrorWithRetry: function() {
 
-      if (!this._isLoading && this._isCacheRunning && !this._fileThrottled) {
-        // storage request initiated from a refresh, Rise Cache is running, and no flag of previous file being throttled
+      var fireError = true;
 
-        if (this._isFile) {
-          fireError = false;
+      if ( !this._hasAttemptedSecondStorageRequest ) {
+        this._hasAttemptedSecondStorageRequest = true;
+        this._loadStorage();
+        fireError = false;
+      }
+      else {
+        if (!this._isLoading && this._isCacheRunning && !this._fileThrottled) {
+          // storage request initiated from a refresh, Rise Cache is running, and no flag of previous file being throttled
 
-          // proceed with getting file from Rise Cache
-          this._getFileFromCache();
-        }
-        else {
-          // folder files
-          if (this._folderFilesToRequest.length > 0) {
+          if (this._isFile) {
             fireError = false;
 
-            // reset the request list/count
-            this._folderFilesRequested = 0;
+            // proceed with getting file from Rise Cache
+            this._getFileFromCache();
+          }
+          else {
+            // folder files
+            if (this._folderFilesToRequest.length > 0) {
+              fireError = false;
+
+              // reset the request list/count
+              this._folderFilesRequested = 0;
 
-            // request the first folder file in the list to get cycle started
-            this._requestCacheFolderFile(this._folderFilesToRequest[0]);
+              // request the first folder file in the list to get cycle started
+              this._requestCacheFolderFile(this._folderFilesToRequest[0]);
+            }
           }
         }
-
       }
 
+      return fireError;
```

Comment:
@rodrigopavezi If this function always returns a value, won't this result in still firing a "rise-storage-error" or "rise-storage-api-error" even though the file(s) will then be attempted to be retrieved from cache and in which case could eventually result in a "rise-storage-response" event also? Or is that intended?